### PR TITLE
lib_chdir:change ret judgment method

### PR DESCRIPTION
**Pull request by anjiahao1**
don't use ret!=0,just use ret
Signed-off-by: anjiahao <anjiahao@xiaomi.com>

### CODE REVIEW

File: libs/libc/unistd/lib_chdir.c [hunk 1]
```diff
@@ -115,7 +115,7 @@ int chdir(FAR const char *path)
   /* Verify that 'path' refers to a directory */
 
   ret = stat(path, &buf);
-  if (ret != 0)
+  if (ret)
```
Comment (pkarashchenko) [inline libs/libc/unistd/lib_chdir.c:118]: could you please explain why `!= 0` is wrong?

Comment (anjiahao1) [reply libs/libc/unistd/lib_chdir.c:118]: ok，i use coverity scan it.coveriy suggest use ret. it does not mean `!=` is wrong 🐰

Comment (pkarashchenko) [reply libs/libc/unistd/lib_chdir.c:118]: maybe `ret < 0` then?

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by pkarashchenko — APPROVED** [review]
LGTM

**Pull request merged**